### PR TITLE
PayId signing implementation

### PR DESCRIPTION
**Pull request by kajoseph**
This a continuation of https://github.com/bitpay/bitcore/pull/2969

What was done:
- Implemented the `sign` function
- Refactored to remove `jose` dependency
- Added tests

### CODE REVIEW

File: packages/bitcore-payid/src/index.ts [hunk 1]
```diff
@@ -1,24 +1,42 @@
...
-  sign(payId: string, address: string, currency: string, identityKey: string): GeneralJWS {
-    // TODO
-    return {} as GeneralJWS;
+  sign(payId: string, address: string, currency: string, identityKey: string | Buffer): GeneralJWS {
+    let jwk = this._convertIdentityKeyToJWK(identityKey);
+
+    const signingParams = new IdentityKeySigningParams(jwk, getDefaultAlgorithm(jwk));
+    const addy = {
+      paymentNetwork: currency,
+      environment: 'mainnet',
```
Comment (matiu) [inline packages/bitcore-payid/src/index.ts:31]: maybe add an optional param 'environment' which defaults to `mainnet`?  so we have the option in the future.

File: packages/bitcore-payid/test/integration/index.spec.ts [hunk 2]
```diff
@@ -118,6 +215,133 @@ describe('PayId', () => {
...
+      const verified = PayId.verify(payId, signed);
+      expect(verified).be.true;
+    });
+
+    it('should work with Bitcore non-HD key', () => {
+      const signed = PayId.sign(payId, addressBTC.addressDetails.address, 'BTC', keys.bitcore.toString());
+      expect(signed).to.exist;
+
+      const verified = PayId.verify(payId, signed);
+      expect(verified).be.true;
```
Comment (matiu) [inline packages/bitcore-payid/test/integration/index.spec.ts:234]: what about adding a tests to PayId.verify with a different payId => verify should by false; with a different signature => verify should be false.

File: packages/bitcore-payid/src/index.ts (hunk 1 above)
Comment (kajoseph) [reply packages/bitcore-payid/src/index.ts]: Done

File: packages/bitcore-payid/test/integration/index.spec.ts (hunk 2 above)
Comment (kajoseph) [reply packages/bitcore-payid/test/integration/index.spec.ts:234]: Good idea

Comment (kajoseph) [reply packages/bitcore-payid/test/integration/index.spec.ts:271]: Done

**Pull request merged**